app/src/main/java/org/p2p/wallet/feerelayer/model/FeeRelayerFee.kt

```diff
@@ -6,27 +6,40 @@ import kotlinx.parcelize.Parcelize
 import org.p2p.solanaj.core.FeeAmount
 
 @Parcelize
-data class FeeRelayerFee constructor(
+data class FeeRelayerFee(
     val transactionFeeInSol: BigInteger,
     val accountCreationFeeInSol: BigInteger,
 
-    val transactionFeeInSpl: BigInteger,
-    val accountCreationFeeInSpl: BigInteger,
+    val transactionFeeInFeePayerToken: BigInteger,
+    val accountCreationFeeInFeePayerToken: BigInteger,
+
+    val transactionFeeInSourceToken: BigInteger,
+    val accountCreateFeeInSourceToken: BigInteger,
 
     val expectedFee: FeeAmount
 ) : Parcelable {
 
-    constructor(feeInSol: FeeAmount, feeInSpl: FeeAmount, expectedFee: FeeAmount) : this(
-        transactionFeeInSol = feeInSol.transactionFee,
-        accountCreationFeeInSol = feeInSol.accountCreationFee,
-        transactionFeeInSpl = feeInSpl.transactionFee,
-        accountCreationFeeInSpl = feeInSpl.accountCreationFee,
+    constructor(
+        feesInSol: FeeAmount,
+        feesInFeePayerToken: FeeAmount,
+        feesInSourceToken: FeeAmount,
+        expectedFee: FeeAmount
+    ) : this(
+        transactionFeeInSol = feesInSol.transactionFee,
```

Comment:
god damn that's a lot of fees

Reply:
yes, adding new them fixes this code https://github.com/p2p-org/key-app-android/pull/2194/files#diff-d782fed09c04c9160bdc93c75c19e0f8a4b12ee5934aa7e4daf3a97c933838c4R171